src/pocketmine/event/block/SignChangeEvent.php

```diff
@@ -73,6 +73,13 @@ public function getLine($index){
 	}
 
 	/**
+	 * @param string[] $lines
+	 */
+	public function setLines(array $lines){
+		$this->lines = $lines;
```

Comment:
What you could do instead is call setLine of the tile. This would automatically provide the out of range information